utils/recursive-lipo

```diff
@@ -73,32 +101,33 @@ def merge_lipo_files(src_root_dirs, file_list, copy_verbatim_subpaths,
                 os.remove(dest_path)
             os.symlink(os.readlink(file_paths[0]), dest_path)
         elif all([os.path.isdir(item) for item in file_paths]):
-            # It's a subdir in all found instances.
-            # See if we should copy verbatim or create the destination subdir.
-            if file in copy_verbatim_subpaths:
-                print("-- Copying subdir verbatim %s" % dest_path)
-                if os.path.isdir(dest_path):
-                    shutil.rmtree(dest_path)
-                shutil.copytree(file_paths[0], dest_path, symlinks=True)
-            else:
-                print("-- Creating subdir %s" % dest_path)
-                if not os.path.isdir(dest_path):
-                    os.makedirs(dest_path)
+            # It's a subdir in all found instances. Create the destination
+            # subdir.
+            print("-- Creating subdir %s" % dest_path)
+            if not os.path.isdir(dest_path):
+                os.makedirs(dest_path)
         elif all([os.path.isfile(item) for item in file_paths]):
             # It's a regular file in all found instances, see if they're
             # identical.
             if all([filecmp.cmp(item, file_paths[0]) for item in file_paths]):
                 # All instances are identical, just copy the unique file.
-                print("-- Copying file %s" % dest_path)
-                shutil.copy2(file_paths[0], dest_path)
+                copy_file(file, src_root_dirs, dest_root_dir)
             elif all([os.access(item, os.X_OK) for item in file_paths]):
-                # Multiple instances are different and executable, try lipo.
-                if verbose:
-                    print("-- Running lipo %s to %s" % (file_paths, dest_path))
+                if (overlapping_lipo_architectures(file_paths,
+                                                   lipo_executable) and
+                    in_any_subpath(file, copy_verbatim_subpaths)):
```

Comment:
From a skim of the code (without running the program), I would be tempt to suggest we may not need `--copy-subdirs` after all

```suggestion
                if overlapping_lipo_architectures(file_paths,
                                                   lipo_executable):
```

In particular, it seems we may end up running `lipo` when (imagine I'm reasoning on a Karnaugh map here)

* the files have at least one overlapping architecture but they are not in a path covered by `--copy-subdirs`

which will lead to lipo failing -- in this sense it should make more sense to copy one of the file verbatim (since they are likely to have the same archs)

As a result, the only condition that determines if we should run lipo or not is whether the executables have overlapping slices or not.

Reply:
This is a good point. `--copy-dirs` is essentially only being used to limit where the overlapping-architectures heuristic is being applied, but this is probably the default behavior we want anyway. I'll create a follow-on PR that removes it completely. Maybe I can get this change to be negative lines-of-code. Thank you!